src/lib.rs

```diff
@@ -27,22 +27,26 @@ use std::env;
 use std::fmt::{self, Display, Formatter, Write};
 use std::fs;
 use std::io::{self, ErrorKind};
+use std::mem::MaybeUninit;
 #[cfg(feature = "fdstore")]
 use std::os::fd::BorrowedFd;
 use std::os::unix::io::RawFd;
 use std::os::unix::net::UnixDatagram;
 use std::process;
 use std::str::FromStr;
 
+use libc::CLOCK_MONOTONIC;
+
 mod ffi;
 
 /// Daemon notification for the service manager.
 #[derive(Clone, Debug)]
 pub enum NotifyState<'a> {
     /// Service startup is finished.
     Ready,
-    /// Service is reloading its configuration. If systemd is v253 or greater, then this message
-    /// MUST be followed by sending a `MonotonicUsec` notification or the reload will fail.
+    /// Service is reloading its configuration.
+    /// On systemd v253 and newer, this message MUST be followed by a
+    /// [`NotifyState::MonotonicUsec`] notification, or the reload will fail.
```

Comment:
Might need to go further and say "the reload will fail and your service will be terminated"

Reply:
Do you know if terminated or restarted? I would expect restarted..

Reply:
The error reports we have received in Kanidm indicate the process is killed, NOT restarted. The bind error report looks to show a similar outcome.